### DISCUSSION TITLE
Ypsomed Pump IoB Reading

### DISCUSSION THREAD

**Discussion by StrangerJona** [NightscoutFoundation/xDrip#3433] [General]
Hey all, I've tried to do a bit of searching and coming up blank. My 2yo daughter has been diagnosed with T1D recently. And has an Ypsomed Pump. My wife who is also a T1D is getting her new pump soon, an Ypsomed Pump too. CamsAPS doesn't have a follow app and uses the same app to interface as a follower or a user. Which will now cause issues as my wife on her phone will have to remove my daughters follow for herself.

I have been playing around with xdrip. And have got it running on my daughters phone to send out out xdrip on both mine and my wife's phone. Which is great for showing her sugars, however, I can't seem to get the app to show IoB despite turning the fetch IoB setting on. The green lines sit at a constant.

Being 2, and going to daycare and all that jazz, my wife is going to miss the ability to see her IoB which the CamsAPS shows.

Question is, is there anything else I can do to get this to work with xdrip on my daughters phone? Its still running the CamsAPS locally and is used for all the bolasing and Dexcom G6 closed loop stuff.

Hopefully this all makes sense. 

Bonus question too. Is there a way xdrip can read the grams bolased when she is eating or having a hypo etc?

**Comment by Navid200** [comment]
xDrip allows one to follow multiple people using variants.  I have explained that here: https://navid200.github.io/xDrip/docs/Variants.html
So, your wife can follow your daughter if she wants to using another installation of xDrip in the form of a variant.  

A Dexcom transmitter has 2 Bluetooth slots.  So that  a medical device, like a pump, and a mobile app, like the Dexcom app, can connect to the transmitter simultaneously.
So, if you were to use a pump that does what it is supposed to do like the t:slim pump, you can use the pump receiving readings from Dexcom transmitter and Dexcom app or xDrip on the other slot.  There are many people who use t:slim pump and xDrip as a collector from G6 or G7.
 
Since CamAPS uses the wrong Bluetooth slot, all of that is messed up. 
As a result, if you use CamAPS, you cannot use xDrip or Dexcom app as a collector.
xDrip developers have added the ability to follow CamAPS.  You can do that using the companion app mode.  But, I don't believe insulin or carbs go through that.
@jamorham can correct me if I am wrong about that last statement.

**Reply by StrangerJona** [reply]
This makes perfect sense as to why it won't do what we'd like it to do.. I guess the next plan will be to approach CamAPS to have them make an actual companion app, or find a way to run 2 instances of a same APK on a phone. The journey continues!

Thankyou greatly for the fast reply too!

**Reply by Navid200** [reply]
To be transparent, I am going to add something here.
When CamAPS did this, a change was made to xDrip hoping for a remedy.  The change was to allow xDrip to switch the Bluetooth slot it uses.  This feature is only available in engineering mode because we have not been able to convince ourselves that it is safe to use.  So, I cannot recommend it.  But, I want to explain what it is just in case you decide to try it, I prefer that you have all the facts.

https://navid200.github.io/xDrip/docs/G6_slot.html

**Comment by feeblee** [comment]
If you and your wife use Android Phones you can use CamAPS FX itself in  Companion Mode. 
See the user manual attached, chapter 9 - Data Upload and Remote Monitoring. In Companion Mode, CamAPS FX should mirror the actual app used for treatment. I never used it myself and this option might not serve your need, as CamAPS FX does not display CoB directly. But if the mirroring works, you should be able to see all treatments in the sideway view.

In case you the Companion Mode doesn't work for you (or you use a iOS device), I provide my current indirect and delayed way to get IoB and CoB into xdrip.
My current setup consists out of several parts, each providing a certain functionality:

**Android Phone** with cellular data plan

- Runs CamAPS FX (Ypsopump, Dexcom G6) and xdrip
- CamAPS FX uploads carb, insulin and bg data to glooko on a hourly basis
- xdrip (bg data) in companion mode catches each bg value and sends it to nightscout
- xdrip (carb and insulin data) downloads carb and insulin data from nightscout

**Nightscout Server**

- receives latest bg reading from xdrip
- receives carb and insulin (both basal and bolus) from glooko upon availability via nightscout-connect

**iPhone** with cellular data plan + AppleWatch

- runs nightguard and downloads the latest bg and treatment data from nightscout upon availabilty


To simplify, the Carb and Insulin data in my setup takes the following path:
CamAPS FX -> Glooko -> Nightscout -> xdrip
The time-wise bottleneck is at Glooko, as some time ago they requested CamAPS to reduce the upload interval.

Hope that helps! 
[user_manual_fx_mmoll_commercial.pdf](https://github.com/NightscoutFoundation/xDrip/files/15299438/user_manual_fx_mmoll_commercial.pdf)

**Reply by BILLYEDIT** [reply]
Hello, and thanks for the info here!   I've now got xdrip and nighscout running for my 4 year old.   Has there been any updates as to what xdrip can read from camAPS?  I've linked Glooko but its no good for up to date IOB etc.  Hoping i've emissed a trick and there is a way of geting more info from camAPS into xdrip?    Kind regards and many thanks. Will

**Reply by feeblee** [reply]
I am not aware of any news regarding the transmission of insulin and carb data. I still use the setup I described. The delay of insulin and carb data still persists, because of the upload intervall from CamAPS FX to Glooko.

**Reply by BossonAnders** [reply]
Solution to follow two persons with CamAPS:

Use a feature called "secure folder" (might be only availabe on Samsungphones). Secure folder enables you to download another instance of CamAPS FX, which enables you to have 2 CamAPS FX. Uses it myself to follow my two t1d-children. 

https://www.samsung.com/uk/support/mobile-devices/what-is-the-secure-folder-and-how-do-i-use-it/

**Reply by feeblee** [reply]
I switched to the CamAPS iOS version from android.
To my knowledge there is no way to capture the glucose readings on iOS.
But I found a solution by using CamAPS as Companion on the android phone. the iOS CamAPS is mirrored on the android phone and xdrip
takes the glucose values from the notification.

So my setup described above was extended by one more step:

CamAPS FX on iOS -> CamAPS FX on android -> xdrip -> Nightscout
